src/subscription.js

```diff
@@ -294,33 +304,41 @@ Subscription.prototype.ack_ = function(message) {
  */
 Subscription.prototype.acknowledge_ = function(ackIds, connId) {
   var self = this;
-  var promise;
 
   ackIds = arrify(ackIds);
 
-  if (!this.isConnected_()) {
-    promise = common.util.promisify(this.request).call(this, {
-      client: 'SubscriberClient',
-      method: 'acknowledge',
-      reqOpts: {
-        subscription: this.name,
-        ackIds,
-      },
-    });
-  } else {
-    promise = new Promise(function(resolve, reject) {
-      self.connectionPool.acquire(connId, function(err, connection) {
-        if (err) {
-          reject(err);
-          return;
-        }
-
-        connection.write({ackIds}, resolve);
+  var promises = chunk(ackIds, MAX_ACK_IDS_PER_REQUEST).map(function(
+    ackIdChunk
+  ) {
+    if (!self.isConnected_()) {
+      return common.util.promisify(self.request).call(self, {
+        client: 'SubscriberClient',
+        method: 'acknowledge',
+        reqOpts: {
+          subscription: self.name,
+          ackIds: ackIdChunk,
+        },
       });
-    });
-  }
+    } else {
```

Comment:
Small nitpick - if we're returning, we shouldn't need this `else` block anymore.

Reply:
Sure, I'm a big fan of early return as well. Wasn't sure if this was the style being used here as well. Changed it!